integration/tests/bft/wintermute/wintermute_test.go

```diff
@@ -0,0 +1,76 @@
+package wintermute
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+
+	"github.com/onflow/flow-go/model/flow"
+)
+
+// wintermuteTimeout corresponds to the timeout the wintermute orchestrator have to conduct the attack.
+const wintermuteTimeout = 1 * time.Minute
+
+type WintermuteTestSuite struct {
+	Suite
+}
+
+func TestWintermuteAttackTestSuite(t *testing.T) {
+	suite.Run(t, new(WintermuteTestSuite))
+}
+
+// TestWintermuteAttack is passing upon a successful wintermute attack is conducted, i.e., wintermute attack orchestrator corrupting an execution
+// result, and then orchestrating its corrupted verification nodes to verify it. The test passes if enough result approvals for corrupted chunks
+// are witnessed in the test network, as well as the block corresponding to the corrupted execution result is sealed.
+func (w *WintermuteTestSuite) TestWintermuteAttack() {
+
+	corruptedResult := w.waitForExecutionResultCorruption()
+	victimBlock := w.BlockState.WaitForBlockById(w.T(), corruptedResult.BlockID)
+
+	// waits for the execution receipt of victim block from both CORRUPTED execution nodes.
+	receiptB1 := w.ReceiptState.WaitForReceiptFrom(w.T(), victimBlock.Header.ID(), w.corruptedEN1Id)
+	w.T().Logf("receipt for victim block generated by execution node-1: %x result ID: %x\n", w.corruptedEN1Id, receiptB1.ExecutionResult.ID())
+	receiptB2 := w.ReceiptState.WaitForReceiptFrom(w.T(), victimBlock.Header.ID(), w.corruptedEN2Id)
+	w.T().Logf("receipt for victim block generated by execution node-2: %x result ID: %x\n", w.corruptedEN2Id, receiptB2.ExecutionResult.ID())
+
+	// makes sure corrupted execution nodes generated the corrupted result for this victim block.
+	require.Equal(w.T(), receiptB1.ExecutionResult.ID(), corruptedResult.ID())
+	require.Equal(w.T(), receiptB2.ExecutionResult.ID(), corruptedResult.ID())
+
+	// waits for at least 2 approvals for all chunks of corrupted result from corrupted verification nodes.
+	for i := 0; i < len(corruptedResult.Chunks); i++ {
+		w.ApprovalState.WaitForTotalApprovalsFrom(w.T(),
+			w.corruptedVnIds,
+			corruptedResult.ID(),
+			uint64(i),
+			2)
+	}
+
+	// waits till victim block gets sealed.
+	w.BlockState.WaitForSealed(w.T(), victimBlock.Header.Height)
```

Comment:
⚠️  This is waiting until **a block** with the same height as the victim block is sealed. It doesn't actually check that the specific block we expect is sealed. 

In general, the `BlockState` tracking is making some assumptions which are no longer safe under Byzantine testing  conditions. [Example here](https://github.com/onflow/flow-go/blob/58f80f66b2119c330d9a0a7e8cf37757087581b1/integration/tests/lib/block_state.go#L66-L71).

For this specific case, we could use the Access API to verify `victimBlock` is sealed:
* assert latest sealed block height is > `victimBlock.Height`
* assert querying block by height returns `victimBlock`